src/activity-logger/background/tab-manager.ts

```diff
@@ -175,6 +182,14 @@ export class TabManager {
             }
         }
     }
+
+    getTabUrls(windowId?: number) {
+        return windowId
```

Comment:
The general idea here is totally correct, but more of a code quality improvement thing: the whole `Array.from(this._tabs.values()).map(tab => tab.url)` thing is repeated twice here. Rather than repeating it, just attach the `.filter` call to the chain afterwards if `windowId` is defined. 

(I know putting the `.map` after the `.filter` may seem like a better idea as then the `.map` can skip filtered-out elements, but it's pretty much inconsequential unless we're dealing with some crazy sized arrays - which we're not here. Though you could still put the `.filter` first without repeating that code if you want, although it may result in less readable code - _debatable_.)

Reply:
I made a change but I put the `.filter` first, hope it's okay :)